src/TwoSlotDisplay.js

```diff
@@ -9,7 +9,11 @@ class TwoSlotDisplay extends Component {
     super(props)
     let day = 0, slot = 0, today = props.date, complex = "Any", floor = "Any";
 
-    day = today.getDay() - 1;
+    if (today.getDay() === 0 || today.getDay() === 6) {
```

Comment:
Just a simple suggestion in programming; call the function once as it was called before.

Once you have the day, replace all instances of `today.getDay()` in your code with day. So there will only be a single function call each time the script is loaded.

Also, I would suggest instead of showing Monday time table according to the current hour, sow it for Monday 8AM. This was it shows for the next time the class starts. I guess that is what we do for time post 5PM as well.

Reply:
Ok cool